middleware/fileserver.go

```diff
@@ -59,6 +61,32 @@ func (fh *fileHandler) serveFile(w http.ResponseWriter, r *http.Request, name st
 			return http.StatusNotFound, nil
 		}
 	}
+
+	// When a request accepts gzipped files and a precompressed file sits
+	// alongside the original, then gzipped file will be served instead.
+	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+		name = filepath.Clean(name)
+		if strings.HasPrefix(name, "/") {
+			name = name[1:]
+		}
+		var buffer bytes.Buffer
+		buffer.WriteString(name)
+		buffer.WriteString(".gz")
+		nameGz := buffer.String()
+		if _, err := os.Stat(nameGz); err == nil {
+			name = nameGz
+			// sets response header so the response_filter.go won't commpress it.
+			w.Header().Set("Content-Encoding", "gzip")
+			if w.Header().Get("Content-Type") == "" {
```

Comment:
I think, at this point you can utilize `FileServer.serveFile` or `fh.Next.ServeHTTP(w, r) // ← adjust the r.URL.Path!` to avoid duplicate code.